### PR TITLE
✨ Implementing new `list_services_paginated` entrypoint in web-api (part 1)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

In PR https://github.com/ITISFoundation/osparc-simcore/pull/6003 we implemented a new `rpc` interface in the `catalog` and connected it with the `rest` api of the `webserver` service. This PR **partially** implements `api.rpc._services.list_services_paginated`. 

Specifically:
- implements repository layer (i.e. `db.repositories.services`)
- implements services layer (i.e. `services.catalog.list_services_paginated`)
     -  only the data from the database is included.
     -  data from image labels is still not included
- connects controller layer (i.e. `api.rpc._services`) to service layer 
- ♻️ Adapting `ServiceGetV2` model in coordination with @odeimaiz 

Following up PRs will cover:
- connection with registry/director
- compatibility rules

## Related issue/s

- Follows up from https://github.com/ITISFoundation/osparc-simcore/pull/6003
- Part of #5964 

## How to test

Driving tests
- services/catalog/tests/unit/with_dbs/test_db_repositories.py
- Manual exploratory test: use `/dev/catalog/services-`
  -  set `WEBSERVER_DEV_FEATURE_ENABLED=1` (i.e. configuration in master deploy)
  - start stack
  - open swagger at `/dev/doc`
  - execute `GET /dev/catalog/services/-/latest`. Expected fast and short response body
  - compare with `GET /catalog/services/latest`. Expected slow and long response body

## Dev-ops checklist
None

### CODE REVIEW

File: packages/models-library/src/models_library/services_authoring.py [hunk 1]
```diff
@@ -7,35 +9,48 @@ class Badge(BaseModel):
...
-    affiliation: str | None = Field(
-        None, examples=["Sense8", "Babylon 5"], description="Affiliation of the author"
-    )
+    affiliation: str | None = Field(None)
+
+    class Config:
+        schema_extra: ClassVar[dict[str, Any]] = {
+            "examples": [
+                {
+                    "name": "Jim Knopf",
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_authoring.py:47]: you know who Jim Knopf is? ;)

File: services/catalog/src/simcore_service_catalog/api/rest/_services.py [hunk 2]
```diff
@@ -115,6 +115,7 @@ async def list_services(
     }
     # Non-detailed views from the services_repo database
     if not details:
+        # TODO: what is this used for?
```
Comment (sanderegg) [inline services/catalog/src/simcore_service_catalog/api/rest/_services.py:118]: this is there to make it way faster as it only returns the key/versions available.

File: services/catalog/src/simcore_service_catalog/db/repositories/services.py [hunk 3]
```diff
@@ -185,7 +158,7 @@ async def get_latest_release(self, key: str) -> ServiceMetaDataAtDB | None:
             result = await conn.execute(query)
             row = result.first()
         if row:
-            return ServiceMetaDataAtDB.from_orm(row)
+            return cast(ServiceMetaDataAtDB, ServiceMetaDataAtDB.from_orm(row))
```
Comment (sanderegg) [inline services/catalog/src/simcore_service_catalog/db/repositories/services.py:161]: that is weird

**Review by sanderegg — APPROVED** [review]
thanks

**Review by odeimaiz — COMMENTED** [review]
👍

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks

**Pull request merged**